files.go

```diff
@@ -81,6 +81,21 @@ func (sb *SectorBuilder) TrimCache(sectorID uint64) error {
 	return nil
 }
 
+func (sb *SectorBuilder) CanCommit(sectorID uint64) (bool, error) {
+	dir, err := sb.sectorCacheDir(sectorID)
+	if err != nil {
+		return false, xerrors.Errorf("getting cache dir: %w", err)
+	}
+
+	ents, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return false, err
+	}
+
+	// TODO: slightly more sophisticated check
+	return len(ents) == 10, nil
```

Comment:
What's the significance of the number 10 here? Is this the quantity of artifacts you expect rust-fil-proofs to have written to the cache dir?

Reply:
If so, I think this check should be written in Rust and exposed through CGO. I think it makes sense to put the cache-is-healthy check very close to the place where we write items to the cache. Those two operations (cache health-checking and cache writing) are tightly coupled, and colocating the operations would make that coupling quite clear.

Reply:
Yeah, we definitely need a `cacheIsHealthy` call from rust. Can you write that up for us @laser ?

Reply:
Yeah, please don't hardcode this number, as the number of files is definitely going to fluctuate and not a stable interface